src/shogun/machine/Machine.cpp

```diff
@@ -176,3 +176,21 @@ std::shared_ptr<LatentLabels> Machine::apply_latent(std::shared_ptr<Features> da
 	error("This machine does not support apply_latent()");
 	return NULL;
 }
+
+std::shared_ptr<Machine> Machine::fit(
+    const std::shared_ptr<Features>& data, const std::shared_ptr<Labels>& lab)
+{
+	require(data, "Features not set!");
+	require(lab, "Labels not set!");
```

Comment:
minor: This should read "No labels provided" as it refers to the arguments, not the members ...